share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+#include <stdio.h>
+
+int f(int arg1) {
+  int return_value;
+  while (arg1 > 0) {
+    if (arg1 == 1) {
+      return_value = 10;
+      printf("Argument equal to 1");
```

Comment:
This should be linked with `nostlib`, so no `printf` please.

I suppose this is just a way to prevent the optimizer to optimize away this call.

Just look into the other files in `x86-64` `for-decompilation` to see how we deal with this.

TLDR: use `__attribute__((weak))` all over all the functions whose call sites you want to prevent the optimizer to mess with or remove.

Reply:
Yeah right, the `printf` was there for the reason you highlighted. Changed it following your suggestion in https://github.com/revng/revng-qa/commit/7d9d32d70292d147d1eec204ac75342c74c3fd68